### PR TITLE
Use a `net.Socket` instead of a plain `EventEmitter` for replaying proxy errors

### PR DESCRIPTION
**Pull request by lpinca**
Fixes: https://github.com/TooTallNate/node-https-proxy-agent/issues/81

### CODE REVIEW

**Comment by lpinca** [comment]
Hmm there is no  `{readable,writable}.destroy()` on Node.js 6.

File: index.js [hunk 1]
```diff
@@ -168,7 +168,16 @@ HttpsProxyAgent.prototype.callback = function connect(req, opts, fn) {
...
 			socket.destroy();
-			socket = new events.EventEmitter();
+			socket = new stream.Duplex({
+				read() {},
+				write(chunk, encoding, callback) {
+					callback();
+				}
+			});
+
+			if (process.versions.modules === '48') {
```
Comment (lpinca) [inline index.js]: I'm not very happy with this. I wonder if it's better to use an actual `Socket`.

Comment (lpinca) [reply index.js]: I've used a new and closed `net.Socket` in f6e253806f0898a8d213b2b3650153ca1527f1f1. Let me know if I should squash it, drop it, or keep it as it.

File: index.js [hunk 2]
```diff
@@ -2,10 +2,11 @@
  * Module dependencies.
  */
 
+var assert = require('assert');
 var net = require('net');
 var tls = require('tls');
 var url = require('url');
-var events = require('events');
+var stream = require('stream');
```
Comment (TooTallNate) [inline index.js]: Not used?

**Comment by TooTallNate** [comment]
This looks great! Thanks for the PR

File: index.js (hunk 2 above)
Comment (TooTallNate) [reply index.js]:
```suggestion
```

File: index.js (hunk 1 above)
Comment (TooTallNate) [reply index.js]: I'll squash when merging the PR, don't worry about that part for now.

**Pull request merged**